pkgs/applications/networking/browsers/firefox/wrapper.nix

```diff
@@ -100,19 +100,21 @@ let
       policiesJson = builtins.toFile "policies.json"
         (builtins.toJSON enterprisePolicies);
 
+      usesNixExtensions = nixExtensions != null;
+
       extensions = builtins.map (a:
         if ! (builtins.hasAttr "extid" a) then
-        throw "extraExtensions has an invalid entry. Missing extid attribute. Please use fetchfirefoxaddon"
+        throw "nixExtensions has an invalid entry. Missing extid attribute. Please use fetchfirefoxaddon"
         else
         a
-      ) extraExtensions;
+      ) (if usesNixExtensions then nixExtensions else []);
 
       enterprisePolicies =
       {
-        policies = {
+        policies = lib.optionalAttrs usesNixExtensions  {
           DisableAppUpdate = true;
         } //
-        {
+        lib.optionalAttrs usesNixExtensions {
           ExtensionSettings = {
             "*" = {
                 blocked_install_message = "You can't have manual extension mixed with nix extensions";
```

Comment:
This message should explain how to disable this enforcement.

Reply:
@kevincox can you make a PR?